google-cloud-storage/src/test/scala/akka/stream/alpakka/googlecloud/storage/scaladsl/GCStorageWiremockBase.scala

```diff
@@ -711,16 +712,21 @@ abstract class GCStorageWiremockBase(_system: ActorSystem, _wireMockServer: Wire
         )
     )
 
-  def mockLargeFileUpload(firstChunkContent: String, secondChunkContent: String, chunkSize: Int): Unit = {
+  def mockLargeFileUpload(firstChunkContent: String, secondChunkContent: String, chunkSize: Int): Unit =
+    mockLargeFileUpload(firstChunkContent, secondChunkContent, chunkSize, None)
+
+  def mockLargeFileUpload(firstChunkContent: String,
+                          secondChunkContent: String,
+                          chunkSize: Int,
+                          metadata: Option[(String, String)]): Unit = {
```

Comment:
I've only allowed one key-value pair as metadata here, purely to make creating the json for this simple, without duplicating the code itself (IE `metadata.toJson.compactPrint`)
I'm happy to take suggestions here, as I can see how it would be confusing to to only be able to add a key-value for metadata, when in reality it's a map. I could duplicate the main code for this one section, as it is really only 2 spray json calls, or could create the json "manually" for a map instead of just the tuple

Reply:
I think the mock should support the Map as well.

> I could duplicate the main code for this one section, as it is really only 2 spray json calls, or could create the json "manually" for a map instead of just the tuple?

Either solution sounds fine to me.

Reply:
cool, I'll sort this now